### PR TITLE
Configure Binaries from PipelineConfiguration

### PR DESCRIPTION
**Pull request by as2388**
Replaces the hard-coded binary-handling that was embedded deep within the pipeline with loops that refer to new PipelineConfiguration options.

Note that this still only allows for one reasons scheme and one binary scheme. In future, we will probably want to move to configurations that supported n of each scheme.

### CODE REVIEW

File: project_redss/analysis_file.py [hunk 1]
```diff
@@ -97,35 +97,36 @@ def generate(user, data, csv_by_message_output_path, csv_by_individual_output_pa
...
+                 plan.binary_code_scheme.get_code_with_id(td[plan.binary_coded_field]["CodeID"]).string_value
+                 for plan in PipelineConfiguration.RQA_CODING_PLANS if plan.binary_code_scheme is not None},
+                Metadata(user, Metadata.get_call_location(), time.time())
+            )
+
+        # Translate the RQA reason codes to matrix values
         matrix_keys = []
 
         for plan in PipelineConfiguration.RQA_CODING_PLANS:
+            # TODO: If this was a list, codes would come out in the order they're in in the schemes, rather than
```
Comment (lukechurch) [inline project_redss/analysis_file.py:128]: I think sorted in the order they're in in the schemes is better than by alpha. E.g. it sorts all of the control codes to the end

File: project_redss/apply_manual_codes.py [hunk 2]
```diff
@@ -25,46 +25,24 @@ def apply_manual_codes(cls, user, data, coda_input_dir):
...
             try:
-                coda_input_path = path.join(coda_input_dir, plan.coda_filename)
                 if path.exists(coda_input_path):
                     f = open(coda_input_path, "r")
                 TracedDataCoda2IO.import_coda_2_to_traced_data_iterable_multi_coded(
                     user, rqa_messages, plan.id_field, {plan.coded_field: plan.code_scheme}, f)
+
+                if plan.binary_code_scheme is not None:
+                    if f is not None:
+                        f.seek(0)
```
Comment (lukechurch) [inline project_redss/apply_manual_codes.py:39]: Could we assert that this is true inside the TracedDataCoda2IO.import_coda_2_to_traced_data_iterable method? I could see some really confusing errors occuring if we ever forget to do this

**Review by lukechurch — APPROVED** [review]

File: project_redss/analysis_file.py (hunk 1 above)
Comment (as2388) [reply project_redss/analysis_file.py:128]: Will address on ADSS

File: project_redss/apply_manual_codes.py (hunk 2 above)
Comment (as2388) [reply project_redss/apply_manual_codes.py:39]: If we forget to seek then core does crash immediately.
But the error message is really confusing, I'll add a card for investigating if we can put a check into core.

**Pull request merged**